### ISSUE TITLE
Crash when specific flag is in history

### ISSUE THREAD

**Issue by allainecactuar** [Bluefissure/MapLinker#13]
Maplinker crashes the game when viewing the records tab when this one specific flag is in the history. I was able to recover by manually deleting all the flags from the config file. I tried adding just this one back that corresponded to the time of my game crash, and was able to reproduce the crash every time. Removing it from the list restored usual functionality.  

```json
{
  "$type": "MapLinker.Configuration, MapLinker",
  "MapLinkerLanguage": 5,
  "ShowTooltips": true,
  "FilterDuplicates": true,
  "UseFloatingWindow": false,
  "UILanguage": "en",
  "MapLinkMessageList": [
    {
      "$type": "MapLinker.Objects.MapLinkMessage, MapLinker",
      "ChatType": 101,
      "Sender": "Einar HansenSargatanas",
      "Text": "The Garlok Eastern La Noscea ( 15.0  , 29.9 ) Z: 0.4 100% Siren",
      "X": 15.041239,
      "Y": 30.024538,
      "Scale": 100.0,
      "TerritoryId": 137,
      "PlaceName": "Eastern La Noscea",
      "RecordTime": "2022-01-21T11:22:30.3131921-08:00"
    }
  ],
  "Recording": true,
  "Teleport": true,
  "SortDesc": true,
  "FilterDupTimeout": 5,
  "MaxRecordings": 100,
  "PrintMessage": false,
  "PrintError": true,
  "FilteredChannels": [],
  "Version": 0
}
```

**Comment by allainecactuar** [comment]
I believe the problem is that `ImGui.TextWrapped` takes a format string:

https://github.com/ocornut/imgui/blob/26e59c8742c82ebfa0a696572b21b0bc9f14d5c6/imgui.h#L482

So if you pass it strings with `%` in them they will be interpreted as C-style string format codes. So when passing user-generated content, you need to be more careful. 

https://github.com/Bluefissure/MapLinker/blob/bfd85606390d1986c03f05e7a417d56e570ea067/MapLinker/Gui/ConfigurationWindow.cs#L187

I'm guessing this can be fixed by replacing that call with: 

```cs
ImGui.TextWrapped("%s", maplinkMessage.Text);
```

But I'm not 100% sure how the layers of magic between the Dalamud Plugin and the ImGui native library work, and I'm not set up to build this plugin so I haven't tested it.

**Comment by Bluefissure** [comment]
> I believe the problem is that `ImGui.TextWrapped` takes a format string:
> 
> https://github.com/ocornut/imgui/blob/26e59c8742c82ebfa0a696572b21b0bc9f14d5c6/imgui.h#L482
> 
> So if you pass it strings with `%` in them they will be interpreted as C-style string format codes. So when passing user-generated content, you need to be more careful.
> 
> https://github.com/Bluefissure/MapLinker/blob/bfd85606390d1986c03f05e7a417d56e570ea067/MapLinker/Gui/ConfigurationWindow.cs#L187
> 
> I'm guessing this can be fixed by replacing that call with:
> 
> ```cs
> ImGui.TextWrapped("%s", maplinkMessage.Text);
> ```
> 
> But I'm not 100% sure how the layers of magic between the Dalamud Plugin and the ImGui native library work, and I'm not set up to build this plugin so I haven't tested it.


Thank you for pointing this out, it happened to me several times when the `%` is parsed as the placeholder when formatting the UI.

I will test it out and make an update later today, together with recent updates.

**Comment by allainecactuar** [comment]
Yeah, in this case the danger here in my particular flag is that there was a `%s` which means the next arg is interpreted as a string pointer and then you get an Access Violation.

I'm glad you saw this right away, I was going to DM you but you have DMs turned off. Don't want folks trying to crash other people on purpose. :)

**Comment by Bluefissure** [comment]
Should be fixed in v1.1.4.0, which is in the https://github.com/goatcorp/DalamudPlugins/pull/1919 of DalamudPlugins repo.

**Comment by Bluefissure** [comment]
It's merged so close this issue for now, feel free to reopen if the problem persists.
